File: internal/wikidata/hierarchy_test.go
package wikidata

import "testing"

func TestHierarchyToJSON(t *testing.T) {
	graph := map[string]hierarchyNode{
		"Q42": {
			Label:      "Douglas Adams",
			InstanceOf: []string{"Q5"},
			SubclassOf: nil,
		},
		"Q5": {
			Label:      "human",
			InstanceOf: nil,
			SubclassOf: []string{"Q729"},
		},
		"Q729": {
			Label:      "mammal",
			InstanceOf: nil,
			SubclassOf: nil,
		},
	}

	rendered := hierarchyToJSON("Q42", graph, 2)
	top, ok := rendered.(map[string]any)
	if !ok {
		t.Fatalf("expected map output, got %T", rendered)
	}
	if _, ok := top["Douglas Adams (Q42)"]; !ok {
		t.Fatalf("expected root key not found in rendered hierarchy: %#v", top)
	}
}

File: test/hierarchy.test.ts
import assert from "node:assert/strict";
import test from "node:test";

import { hierarchyToJSON } from "../src/client.js";

test("hierarchyToJSON", () => {
  const graph = {
    Q42: {
      label: "Douglas Adams",
      instanceOf: ["Q5"],
      subclassOf: [],
    },
    Q5: {
      label: "human",
      instanceOf: [],
      subclassOf: ["Q729"],
    },
    Q729: {
      label: "mammal",
      instanceOf: [],
      subclassOf: [],
    },
  };

  const rendered = hierarchyToJSON("Q42", graph, 2);

  assert.equal(typeof rendered, "object");
  assert.ok(rendered && !Array.isArray(rendered));

  const top = rendered as Record<string, unknown>;
  assert.ok(Object.hasOwn(top, "Douglas Adams (Q42)"));
});
